lib/prawn/graphics/blend_mode.rb

```diff
@@ -0,0 +1,78 @@
+# encoding: utf-8
+#
+# blend_mode.rb : Implements blend modes
+#
+# Contributed by John Ford. October, 2015
+#
+# This is free software. Please see the LICENSE and COPYING files for details.
+#
+
+module Prawn
+  module Graphics
+    # The Prawn::BlendMode module is used to change the way
+    # two layers are blended together.
+    #
+    # Passing an array of blend modes is allowed. PDF viewers should
+    # blend layers based on the first recognized blend mode.
+    #
+    # Valid blend modes in v1.4 of the PDF spec include :Normal, :Multiply, :Screen,
+    # :Overlay, :Darken, :Lighten, :ColorDodge, :ColorBurn, :HardLight, :SoftLight,
+    # :Difference, and :Exclusion.
+    #
+    # Example:
+    #   pdf.fill_color('0000ff')
+    #   pdf.fill_rectangle([x, y+25], 50, 50)
+    #   pdf.blend_mode(:Multiply) do
+    #     pdf.fill_color('ff0000')
+    #     pdf.fill_circle([x, y], 25)
+    #   end
+    #
+    module BlendMode
+      BLEND_MODES = [:Normal, :Multiply, :Screen, :Overlay, :Darken, :Lighten, :ColorDodge, :ColorBurn, :HardLight, :SoftLight, :Difference, :Exclusion]
```

Comment:
Or validate the inclusion of `blend_mode` in `BLEND_MODES`?  I feel happy with either solution.